From fd5172e1a0d635d583f765980058ff54e64a1d77 Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Fri, 25 Aug 2023 09:40:13 +0200
Subject: [PATCH 0434/1141] dt-bindings: st,stm32-syscon: add properties for
 st,stm32-tamp-nvram

st,stm32-tamp-nvram requires some specific properties from its parent
st,stm32-syscon.

Change-Id: I05f9807afdf105e0701a5042cc125eb9ceb6e725
---
 .../bindings/arm/stm32/st,stm32-syscon.yaml   | 47 +++++++++++++++----
 1 file changed, 38 insertions(+), 9 deletions(-)

--- a/Documentation/devicetree/bindings/arm/stm32/st,stm32-syscon.yaml
+++ b/Documentation/devicetree/bindings/arm/stm32/st,stm32-syscon.yaml
@@ -34,19 +34,48 @@ properties:
   clocks:
     maxItems: 1
 
+  '#address-cells':
+    const: 1
+
+  '#size-cells':
+    const: 1
+
+  ranges: true
+
+  reboot-mode:
+    type: object
+    $ref: /schemas/power/reset/reboot-mode.yaml#
+
+patternProperties:
+  "^nvram@[0-9a-f]+$":
+    type: object
+    $ref: /schemas/nvmem/st,stm32-tamp-nvram.yaml#
+
 required:
   - compatible
   - reg
 
-if:
-  properties:
-    compatible:
-      contains:
-        enum:
-          - st,stm32mp157-syscfg
-then:
-  required:
-    - clocks
+allOf:
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - st,stm32mp157-syscfg
+    then:
+      required:
+        - clocks
+  - if:
+      properties:
+        compatible:
+          contains:
+            enum:
+              - st,stm32-tamp
+    then:
+      required:
+        - '#address-cells'
+        - '#size-cells'
+        - ranges
 
 additionalProperties: false
 
